src/EnergyPlus/DataSizing.cc

```diff
@@ -1094,4 +1317,64 @@ Real64 OARequirementsData::calcOAFlowRate(EnergyPlusData &state,
     return OAVolumeFlowRate;
 }
 
+Sched::Schedule *OARequirementsData::getZoneFlowFracSched(EnergyPlusData &state, bool notAllSame)
+{
+    notAllSame = false;
+    Sched::Schedule *schedPtr = nullptr;
+    if (this->numDSOA == 0) {
+        // This is a simple DesignSpecification:OutdoorAir
+        schedPtr = this->oaFlowFracSched;
+    } else {
+        // This is a DesignSpecification:OutdoorAir:SpaceList
+        for (int dsoaCount = 1; dsoaCount <= this->numDSOA; ++dsoaCount) {
+            auto const &thisDSOA = state.dataSize->OARequirements(this->dsoaIndexes(dsoaCount));
+            if (dsoaCount == 1) {
+                schedPtr = thisDSOA.oaFlowFracSched;
+            } else {
+                if (schedPtr != thisDSOA.oaFlowFracSched) {
+                    notAllSame = true;
+                    ShowWarningError(state,
+                                     format("getZoneFlowFracSched: Outdoor Air Schedules are not the same for all spaces in "
+                                            "DesignSpecification:OutdoorAir:SpaceList={}.",
+                                            this->Name));
+                    ShowContinueError(state, format("Using the first space schedule={}", schedPtr->Name));
```

Comment:
If the spaces in the DSOA:SpaceList do not all use the same schedule, then throw a warning and return the first one, because contaminants are currently only tracked at the zone level. Should probably extend those to spaces (someday).